### PR TITLE
Add some check about image garbagecollect

### PR DESCRIPTION
**Pull request by zhengguoyong**

### CODE REVIEW

File: pkg/kubelet/image_manager.go [hunk 1]
```diff
@@ -101,6 +101,9 @@ func newImageManager(runtime container.Runtime, cadvisorInterface cadvisor.Inter
 	if policy.LowThresholdPercent < 0 || policy.LowThresholdPercent > 100 {
 		return nil, fmt.Errorf("invalid LowThresholdPercent %d, must be in range [0-100]", policy.LowThresholdPercent)
 	}
+	if policy.LowThresholdPercent > policy.HighThresholdPercent {
+		return nil, fmt.Errorf("invalid LowThresholdPercent %d, must be small than HighThresholdPercent %d", policy.LowThresholdPercent, policy.HighThresholdPercent)
```
Comment (derekwaynecarr) [inline pkg/kubelet/image_manager.go]: must be smaller than

File: pkg/kubelet/image_manager.go [hunk 2]
```diff
@@ -204,15 +207,16 @@ func (im *realImageManager) GarbageCollect() error {
...
-			err := fmt.Errorf("failed to garbage collect required amount of images. Wanted to free %d, but freed %d", amountToFree, freed)
-			im.recorder.Eventf(im.nodeRef, api.EventTypeWarning, container.FreeDiskSpaceFailed, err.Error())
-			return err
+		if amountToFree > 0 {
+			freed, err := im.freeSpace(amountToFree)
+			if err != nil {
+				return err
+			}
+			if freed < amountToFree {
+				err := fmt.Errorf("failed to garbage collect required amount of images. Wanted to free %d, but freed %d", amountToFree, freed)
```
Comment (derekwaynecarr) [inline pkg/kubelet/image_manager.go]: the units here appear to be bytes.

should the error read instead:

"failed to garbage collect required amount of bytes.  wanted to free %d, but freed %d"

File: pkg/kubelet/image_manager.go (hunk 1 above)
Comment (zhengguoyong) [inline pkg/kubelet/image_manager.go]: Done.

File: pkg/kubelet/image_manager.go (hunk 2 above)
Comment (zhengguoyong) [inline pkg/kubelet/image_manager.go]: Done.

**Comment by zhengguoyong** [comment]
@derekwaynecarr 
thanks so much for your review. :)
i have changed the code based on your view.

**Comment by zhengguoyong** [comment]
@k8s-bot  test this please

File: pkg/kubelet/image_manager.go [hunk 3]
```diff
@@ -207,15 +207,16 @@ func (im *realImageManager) GarbageCollect() error {
...
-		freed, err := im.freeSpace(amountToFree)
-		if err != nil {
-			return err
-		}
-
-		if freed < amountToFree {
-			err := fmt.Errorf("failed to garbage collect required amount of images. Wanted to free %d, but freed %d", amountToFree, freed)
-			im.recorder.Eventf(im.nodeRef, api.EventTypeWarning, container.FreeDiskSpaceFailed, err.Error())
-			return err
+		if amountToFree > 0 {
```
Comment (vishh) [inline pkg/kubelet/image_manager.go]: Why will this check be necessary? Unless `LowThresholdPercent` == `HighThresholdPercent`, this check isn't necessary right? Why would someone do that? We should fail validation in such scenarios.

**Comment by vishh** [comment]
Whats the reason for posting this PR?

**Comment by zhengguoyong** [comment]
In fact , this PR has another chages.  similar as #18073. aim to check LowThresholdPercent  > HighThresholdPercent .
As it is merged in PR #18073, so when i rebase, it remain what you can see now.

**Comment by vishh** [comment]
@zhengguoyong Can we close this PR? If you disagree, can you clarify what you are trying to fix?

**Pull request closed without merging**